### PR TITLE
feat: check for minimum anvil version

### PR DESCRIPTION
**Pull request by tremarkley**
Closes https://github.com/ethereum-optimism/supersim/issues/38

Implements a check that the machine has a minimum anvil version of 0.2.0 installed prior to starting supersim.

### CODE REVIEW

**Comment by tremarkley** [comment]
* **#47** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/supersim/47?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tremarkley and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: cmd/main.go [hunk 1]
```diff
@@ -59,3 +73,53 @@ func SupersimMain(ctx *cli.Context, closeApp context.CancelCauseFunc) (cliapp.Li
...
+
+	if isVersionGreaterOrEqual(version, minAnvilVersion) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// Compares two semantic versions.
+func isVersionGreaterOrEqual(version, minVersion string) bool {
```
Comment (jakim929) [inline cmd/main.go:105]: nit: I think this package has the functionality you're looking for https://pkg.go.dev/golang.org/x/mod/semver
```
func Compare(v, w string)
```

**Review by jakim929 — CHANGES_REQUESTED** [review]
foundry doesn't follow semver until 1.0.0 is released 
https://github.com/foundry-rs/foundry/blob/5b8416363c7c1aab6d3cda0da6e7c65594ad8a12/CHANGELOG.md

so 0.2.0 has been around for more than a year and covers a huge range of releases.

`foundryup` installs the latest nightly release. It might make more sense to check that the timestamp on the nightly release is past a certain minimum version.
`anvil 0.2.0 (0116be1 2024-07-09T00:18:45.485630000Z)`

Not sure what the minimum version we are trying to enforce is though.

File: cmd/main.go [hunk 2]
```diff
@@ -45,6 +54,11 @@ func main() {
 
 func SupersimMain(ctx *cli.Context, closeApp context.CancelCauseFunc) (cliapp.Lifecycle, error) {
 	log := oplog.NewLogger(oplog.AppOut(ctx), oplog.ReadCLIConfig(ctx))
+	ok, minAnvilErr := isMinAnvilInstalled(log)
+
+	if !ok || minAnvilErr != nil {
+		return nil, fmt.Errorf("anvil version %s or higher is required.", minAnvilVersion)
```
Comment (jakim929) [inline cmd/main.go:60]: nit: would be nice to also print `please use foundryup to update to the latest version

File: cmd/main.go (hunk 1 above)
Comment (jakim929) [reply cmd/main.go:105]: prob ignore this if we're not going with semver

**Review by jakim929 — APPROVED** [review]

**Pull request merged**